### PR TITLE
feat(icon): add some new icons

### PR DESCRIPTION
**Pull request by 321gillian**
Add some new icons

GDS-2374

--- 
Updated to deprecate arrow-left-to-line-solid and arrow-right-to-line-solid. These icons were added in December. They ended up being the wrong version of what the requestor was looking for. Design are seeking to deprecate these now to reflect the mistake in adding them. 

They don't need to map to anything else as the name attached is a meaningful name in font-awesome and mapping would be confusing for the team working with them. 

I've re-added the getRootIconName function from v3 to support this so can point the current name to file in legacy folder.

### CODE REVIEW

**Comment by katie-bobbe-genesys** [comment]
These icon updates look good to me, just resolve the conflict in the gux-icon readme file and I'll approve it

File: packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.types.ts [hunk 1]
```diff
@@ -183,9 +183,9 @@ export type GuxIconIconName =
   | 'fa/align-right-regular'
   | 'fa/arrow-down-left-and-arrow-up-right-to-center-regular'
   | 'fa/arrow-down-wide-short-regular'
-  | 'fa/arrow-left-to-line-solid'
+  | 'fa/arrow-left-to-line-regular'
   | 'fa/arrow-right-to-bracket-regular'
-  | 'fa/arrow-right-to-line-solid'
+  | 'fa/arrow-right-to-line-regular'
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.types.ts:188]: We will have to update the script that generates this type. It should include:

- 'fa/arrow-left-to-line-solid'
- 'fa/arrow-right-to-line-solid'
- 'legacy/fa/arrow-left-to-line-solid'
- 'legacy/fa/arrow-right-to-line-solid'

File: web-apps/genesys-spark-examples/src/utils/generateFontAwesomeIcons.js [hunk 2]
```diff
@@ -36,9 +36,9 @@ const icons = [
       'fa/align-right-regular',
       'fa/arrow-down-left-and-arrow-up-right-to-center-regular',
       'fa/arrow-down-wide-short-regular',
-      'fa/arrow-left-to-line-solid',
+      'fa/arrow-left-to-line-regular',
       'fa/arrow-right-to-bracket-regular',
-      'fa/arrow-right-to-line-solid',
+      'fa/arrow-right-to-line-regular',
```
Comment (daragh-king-genesys) [inline web-apps/genesys-spark-examples/src/utils/generateFontAwesomeIcons.js:41]: I think it is fine not to display the legacy icons here

File: packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.types.ts (hunk 1 above)
Comment (321gillian) [reply packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.types.ts:188]: done

Comment (daragh-king-genesys) [reply packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.types.ts:188]: 'fa/arrow-left-to-line-solid'
'fa/arrow-right-to-line-solid'
are still missing.

Comment (321gillian) [reply packages/genesys-spark-components/src/components/stable/gux-icon/gux-icon.types.ts:189]: Fixed

**Review by daragh-king-genesys — APPROVED** [review]

**Pull request merged**